tensorflow/core/framework/tensor.cc

```diff
@@ -677,14 +677,42 @@ bool Tensor::CanUseDMA() const {
 #undef CASE
 
 namespace {
+//from left dim to right dim , recursive print
 template <typename T>
-string SummarizeArray(int64 limit, int64 num_elts, const char* data) {
+void PrintOneDim(T *data, int dim_index, gtl::InlinedVector<int64, 4> shape,
```

Comment:
See below for parameter order:
dim_indx, shape, limit, shape_size, data, data_index, result.